src/modules/Writers.jl

```diff
@@ -59,13 +59,91 @@ function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNodes, page,
     end
 end
 
+"""
+Converts the function argument tuple `(name, type)` into a string.
+
+The tuple comes from the second return element of the `Base.arg_decl_parts(::Method)`
+and it seems they are always both `::String` (`::ASCIIString` in 0.4).
+It also appears that if the type is not declared for the method, `arg_decl_parts`
+returns an empty string.
+
+The returned string is `name::type` or just `name`, if the type is not declared.
+"""
+function join_decl(decl)
+    n, t = decl
+    isempty(t) ? n : "$(n)::$(t)"
+end
+
 function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNode, page, doc)
     # Docstring header based on the name of the binding and it's category.
     anchor = "<a id='$(node.anchor.id)' href='#$(node.anchor.id)'>#</a>"
     header = "**`$(node.object.binding)`** &mdash; *$(Utilities.doccat(node.object))*."
     println(io, anchor, "\n", header, "\n\n")
     # Body. May contain several concatenated docstrings.
     renderdoc(io, mime, node.docstr, page, doc)
+
+    # Table of methods.
+    # If DocsNode.methods is nulled, then we assume that we should not render a
+    # table. However, if the list of methods is there but has no elements then
+    # we output an appropriate note saying that the name has no methods associated
+    # with it.
+    Utilities.unwrap(node.methods) do ms
+        println(io, "<p><strong>Methods</strong></p>")
+
+        if isempty(ms)
+            println(io, "<p><emph>This name has no methods.</emph></p>")
```

Comment:
This is a good question about terminology actually, also ties into "constructor" vs "method" discussion a little bit.

My own impression based on what is see in the julia documentation is that you can refer to every item in a module collectively as a "name" (from [`methods()`](http://docs.julialang.org/en/latest/stdlib/base/?highlight=names#Base.names)). Also, [constructors are technically functions](http://docs.julialang.org/en/latest/manual/constructors/), which can have several methods associated with them. So, does the type have methods or the function associated with the type? Or are methods associated with the names?

Regarding constructor vs method, again the docs refer to different constructors as methods. And you could (although probably shouldn't) write a "constructor" that returns an object of a different type, so I would stick to calling them methods.

Basically, as a philosophy, I would stick with the terms that are technically correct, rather than making assumptions about the user's intentions (e.g. not assume that all type methods are constructors), even if that goes against widespread practices. But there's still the matter of figuring out what is technically correct.